### PR TITLE
[FLINK-23845][docs]improve PushGatewayReporter config:deleteOnShutdown  de…

### PR DESCRIPTION
**Pull request by camilesing**
the problem takes my team a lot of time... until later, we find the issue https://issues.apache.org/jira/browse/FLINK-13787.  whatever the problem has always existed, we should avoid other guys met it, so I want to update the doc,  do you think about it 

Jira issue link:https://issues.apache.org/jira/browse/FLINK-23845

### CODE REVIEW

**Comment by camilesing** [comment]
@rmetzge  can you review it? thx

**Comment by camilesing** [comment]
@tsreaper  hi, can you review it? thx

File: docs/layouts/shortcodes/generated/prometheus_push_gateway_reporter_configuration.html
Comment (tsreaper) [inline docs/layouts/shortcodes/generated/prometheus_push_gateway_reporter_configuration.html:15]: I'd like to emphasize in the docs that this config only "tries" to delete the metrics, but it is not guaranteed. For example

> Specifies whether to delete metrics from the PushGateway on shutdown. Flink will try its best to delete the metrics but this is not guaranteed. See <a href="https://issues.apache.org/jira/browse/FLINK-13787">here</a> for more details.

**Review by tsreaper — COMMENTED** [review]
Thanks for contributing to Flink! I've left a comment.

File: docs/layouts/shortcodes/generated/prometheus_push_gateway_reporter_configuration.html
Comment (camilesing) [reply docs/layouts/shortcodes/generated/prometheus_push_gateway_reporter_configuration.html:15]: thx, that's a good suggestion

File: flink-metrics/flink-metrics-prometheus/src/main/java/org/apache/flink/metrics/prometheus/PrometheusPushGatewayReporterOptions.java
Comment (tsreaper) [inline flink-metrics/flink-metrics-prometheus/src/main/java/org/apache/flink/metrics/prometheus/PrometheusPushGatewayReporterOptions.java:63]: The link should be attached to "here", not to "for more details".

Comment (camilesing) [reply flink-metrics/flink-metrics-prometheus/src/main/java/org/apache/flink/metrics/prometheus/PrometheusPushGatewayReporterOptions.java]: got it, thx

File: docs/layouts/shortcodes/generated/prometheus_push_gateway_reporter_configuration.html
Comment (tsreaper) [inline docs/layouts/shortcodes/generated/prometheus_push_gateway_reporter_configuration.html:15]: The html file is not in sync with PrometheusPushGatewayReporterOptions.java

**Comment by tsreaper** [comment]
Also please don't squash commits and force-push them, as it will be hard for reviewers to see what is changed. The committer who merges this commit will squash them for you.

File: docs/layouts/shortcodes/generated/prometheus_push_gateway_reporter_configuration.html
Comment (camilesing) [reply docs/layouts/shortcodes/generated/prometheus_push_gateway_reporter_configuration.html]: thx, I'm too careless

**Comment by camilesing** [comment]
@tsreaper got it

**Review by tsreaper — APPROVED** [review]
Looks good to me. Let's wait for the CI to pass.

**Comment by tsreaper** [comment]
Hi @camilesing , the CI is failing. It might be that the code base in your PR has some issues. Please rebase your changes on the latest master branch and force push it to re-trigger the CI.

**Review by tillrohrmann — APPROVED** [review]
Thanks for this contribution @camilesing and the good review @tsreaper. LGTM. Merging this PR now.

**Comment by camilesing** [comment]
@tsreaper I just saw it, but it seems ok now. thanks for your reminder!

**Pull request closed without merging**
